dbms/src/Common/FiberRWLock.h

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <boost/fiber/all.hpp>
+
+namespace DB
+{
+/// FiberRWLock has the same interface with `std::shared_mutex` and is intended to
```

Comment:
Could you specific which type of the RWLock in the annotation, eg: Read first or Write first.

Reply:
good idea.